assets/shared/components/modal/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, close as closeIcon } from '@wordpress/icons';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Modal component.
+ *
+ * @param {Object}   props
+ * @param {Function} props.onClose  Callback to run when trying to close the modal.
+ * @param {string}   props.title    The title for the modal. Empty by default.
+ * @param {Object}   props.children The content of the modal.
+ * @return {JSX.Element} The modal component or null (if not open).
+ */
+const Modal = ( { onClose, title = '', children } ) => {
+	useRunOnEscape( onClose );
+
+	return (
+		<div className={ 'sensei-modal' }>
+			<button
+				className="sensei-modal sensei-modal__overlay"
+				aria-label="Close"
+				onClick={ onClose }
+			/>
+			<div className="sensei-modal__wrapper">
+				<div className="sensei-modal__header">
+					<div className="sensei-modal__title">{ title }</div>
+					<button
```

Comment:
I think we can add the `aria-label` here to this button to make it accessible.

Reply:
Done in da7e3965d